### PR TITLE
feat: add options allow disable automigrate

### PR DESCRIPTION
**Pull request by warjiang**
background:
gorm-adapter will automigrate table,  this pr will provide some options allow developer disable automigrate feature for some special situation

feature:
1. add adapter options
2. allow disable automigrate

### CODE REVIEW

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: options.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package gormadapter
```
[the remark this answers is not in the document]
Comment (warjiang) [reply options.go:15]: done

File: adapter.go [hunk 2]
```diff
@@ -181,14 +182,18 @@ func NewAdapter(driverName string, dataSourceName string, params ...interface{})
 
 // NewAdapterByDBUseTableName creates gorm-adapter by an existing Gorm instance and the specified table prefix and table name
 // Example: gormadapter.NewAdapterByDBUseTableName(&db, "cms", "casbin") Automatically generate table name like this "cms_casbin"
-func NewAdapterByDBUseTableName(db *gorm.DB, prefix string, tableName string) (*Adapter, error) {
+func NewAdapterByDBUseTableName(db *gorm.DB, prefix string, tableName string, opts ...Option) (*Adapter, error) {
```
[the remark this answers is not in the document]
Comment (warjiang) [reply adapter.go:185]: i have add usage for `Option`, another question, should move `tablename`、`prefix` into `Option`, and change the ussage of `NewAdapterByDBUseTableName ` to 
```
// before
a, _ := gormadapter. NewAdapterByDBUseTableName(db, "{{your_table_prefix}}", "{{your_table_name}}")

// after
a, _ := gormadapter. NewAdapterByDBUseTableName(db, 
    gormadapter.WithPrefix("{{your_table_prefix}}"),
    gormadapter.WithTableName("{{your_table_name}}"),
    gormadapter.WithAutoMigrate("{{true or false}}"),
)

```

[the remark this answers is not in the document]
Comment (hsluoyz) [reply adapter.go:185]: @warjiang no breaking change.

File: options.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 The casbin Authors. All Rights Reserved.
```
Comment (hsluoyz) [inline options.go:1]: 2021

File: adapter.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (warjiang) [reply adapter.go:185]: > @warjiang no breaking change.

got it.

File: options.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+package gormadapter
+
+type adapterOptions struct {
+	autoMigrate bool
+}
+
+type Option func(options *adapterOptions)
+
+func defaultAdapterOptions() adapterOptions {
```
Comment (hsluoyz) [inline options.go:23]: Don't use function pointer, hard to read

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@warjiang I think it's a very simple change, but the code here looks too difficult. So sorry closed here. Maybe one of our members will finish this.

**Pull request closed without merging**